### ISSUE TITLE
service instance that is being updated has old plan ID

### ISSUE THREAD

**Issue by Dagrol** [cloudfoundry/cloud_controller_ng#649]
After changing the plan of a service instance with `cf update-service some-instance -p new-plan`, and the service broker chooses to update the service instance asynchronously, the instance is in `update in progress` state.

If we hit the [API endpoint to retrieve the service instance](https://apidocs.cloudfoundry.org/239/service_instances/retrieve_a_particular_service_instance.html), we see the `service_plan_url` is for old plan. Similarly, if we [list instances for the old plan](https://apidocs.cloudfoundry.org/239/service_plans/list_all_service_instances_for_the_service_plan.html), the instance that is being updated appears.

The instance is only "transitioned" to the new plan once the update is complete.

This was unexpected to us, as fresh instances that are `create in progress` (when being provisioned using the async broker api) appear in their plan's list of instances, and have a plan URL.

We are trying to use Cloud Controller as a source of truth for service instances, to avoid having a database for the on-demand broker, and this sort of thing affects the way we implement quotas.

It would be great to hear your thoughts!

cc @avade @craigfurman

**Comment by avade** [comment]
@SocalNick can we also discuss this today.

**Comment by avade** [comment]
Nick will be talking to the CAPI team about this soon.

**Comment by SocalNick** [comment]
CAPI - when you get to this, let's chat. I have a feeling we don't record the new plan until after update because update could fail. It's interesting that CC is mostly concerned with desired state for apps but cannot do the same for updating a service instance.

**Comment by SocalNick** [comment]
@Dagrol @avade we don't have much more to report on this. it's not unreasonable, but seems like it would take a bit of effort and it's not our priority at the moment. could you propose something simple like extending the last_operation that is returned with `/v2/service_instances/:guid`?

```
cf curl /v2/service_instances/391d4db0-1e82-4695-931a-033bb95c6ecc
{
   "metadata": {
      "guid": "391d4db0-1e82-4695-931a-033bb95c6ecc",
      "url": "/v2/service_instances/391d4db0-1e82-4695-931a-033bb95c6ecc",
      "created_at": "2016-09-19T22:59:18Z",
      "updated_at": null
   },
   "entity": {
      "name": "fake-async",
      "credentials": {},
      "service_plan_guid": "bb06938f-e9ac-4817-b0f6-17efefd99d5b",
      "space_guid": "cfdcbe8e-788a-440b-b191-9f1054660ac0",
      "gateway_data": null,
      "dashboard_url": null,
      "type": "managed_service_instance",
      "last_operation": {
         "type": "update",
         "state": "in progress",
         "description": "",
         "updated_at": "2016-09-19T23:17:39Z",
         "created_at": "2016-09-19T23:17:39Z"
      },
      "tags": [],
      "space_url": "/v2/spaces/cfdcbe8e-788a-440b-b191-9f1054660ac0",
      "service_plan_url": "/v2/service_plans/bb06938f-e9ac-4817-b0f6-17efefd99d5b",
      "service_bindings_url": "/v2/service_instances/391d4db0-1e82-4695-931a-033bb95c6ecc/service_bindings",
      "service_keys_url": "/v2/service_instances/391d4db0-1e82-4695-931a-033bb95c6ecc/service_keys",
      "routes_url": "/v2/service_instances/391d4db0-1e82-4695-931a-033bb95c6ecc/routes"
   }
}
```

**Comment by SocalNick** [comment]
Closing due to inactivity
